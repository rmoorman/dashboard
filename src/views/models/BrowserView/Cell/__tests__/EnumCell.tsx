import * as React from 'react' // tslint:disable-line
import EnumCell from '../EnumCell'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

test('EnumCell renders', () => {

  const save = jest.fn((value: TypedValue) => { /* */ })
  const cancel = jest.fn()
  const onKeyDown = jest.fn()
  const field = {
    id: 'cip3p48sj001e1jsmghwkdt2k',
    name: 'description',
    description: '',
    isReadonly: true,
    isList: false,
    isSystem: true,
    typeIdentifier: 'Enum',
    enumValues: ['asd', 'def']
    relatedModel: null,
  }

  const component = shallow(
    <EnumCell
      value='asd'
      save={save}
      cancel={cancel}
      onKeyDown={onKeyDown}
      field={field}
    />
  )

  expect(shallowToJson(component)).toMatchSnapshot()

})
