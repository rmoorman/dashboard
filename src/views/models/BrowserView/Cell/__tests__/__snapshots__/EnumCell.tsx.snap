exports[`test EnumCell renders 1`] = `
<select
  autoFocus={true}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  value="asd">
  <option
    disabled={true}>
    Select an Enum â–¾
  </option>
  <option>
    asd
  </option>
  <option>
    def
  </option>
</select>
`;
